x-pack/plugins/security_solution/server/lib/detection_engine/signals/signal_rule_alert_type.ts

```diff
@@ -413,5 +415,21 @@ export const signalRulesAlertType = ({
         });
       }
     },
+    errorCB: async ({ alertId, message }) => {
+      logger.error(`Detections rule with alert Id: ${alertId} failed with message: ${message}`);
+      const savedObjectsClient = await errorService;
+      if (savedObjectsClient != null) {
+        const ruleStatusClient = ruleStatusSavedObjectsClientFactory(
+          savedObjectsClient as SavedObjectsRepository
```

Comment:
Rather than this `as cast` it looks like we can just modify the ruleStatusSavedObjectsClientFactory like so?

```ts
export const ruleStatusSavedObjectsClientFactory = (
  savedObjectsClient: SavedObjectsClientContract | SavedObjectsRepository | ISavedObjectsRepository
): RuleStatusSavedObjectsClient => ({
  find: (options) =>
    savedObjectsClient.find<IRuleStatusAttributes>({ ...options, type: ruleStatusSavedObjectType }),
  create: (attributes) => savedObjectsClient.create(ruleStatusSavedObjectType, attributes),
  update: (id, attributes) => savedObjectsClient.update(ruleStatusSavedObjectType, id, attributes),
  delete: (id) => savedObjectsClient.delete(ruleStatusSavedObjectType, id),
});
```

That seemed to work for me without conflicts and would be better than an as cast IMO. Totally optional though.

Reply:
Yup that looks good to me! Thanks. I will update.